core/src/engine/io/tls_wrapper_benchmark.cpp

```diff
@@ -0,0 +1,223 @@
+PDKBSbGFQNhq6gcDbA33/Zkh3QvsDxUif5lRSSKEmXbYiTuoo9mUDQ8qKsb4T5v6
+uMovOi0d77cCgYEAjlnncJJ4xzkS6gE8qhBrOnjN3UbIZanAAfB9LdbYaYLEq0rZ
+SEPmVSKqNWPpmTvowrxoP2oih5z23D3CUsCxT/uEAW0JsULo0M1dvNadRTbscwLc
+eGO+/PoCe7bv3GD37U2tdxzL69n9wWMuhU/ltJnkj/GKpskZkPAMX4t+Bs0CgYAf
+ln+AkhIul6fzJP2t41SXIbM2NtbVNJjjG8kjWQiUCM8Idta4wOdyXx9MTsFLLvZ0
+8jabg/UER9kFqdQnWcrjSnqwMt5SDdTbxEuvzc6Gxs/9ufYK3MKTboRxyNCZpSQW
+IuZxTtatFjYu12bTmdoqKl2MZEkOf35lhfY848maawKBgACxJXDPWx/Ah4cT0LbN
+XFdWxjSVMU2BHlO0rWy8oZwFaOH8cy+dYe1cdMh+nQLMNXXTY+nu4WditpLfURof
+2GmF9LD5e6tUJw1rPLbKDVM5DXuO3ka0msRea4Y8Diaiql8eLytJhbMw12nE0B1g
+LKaaQekBQ5eebwM3oRc68q0K
+-----END PRIVATE KEY-----)";
+
+constexpr auto other_cert = R"(-----BEGIN CERTIFICATE-----
+MIIDITCCAgmgAwIBAgIUL5c8L0jabF9rZ2w383zGQmwcLZwwDQYJKoZIhvcNAQEL
+BQAwIDEeMBwGA1UEAwwVdGxzd3JhcHBlcl90ZXN0X290aGVyMB4XDTIyMDIxNzEy
+MjcxOVoXDTMyMDIxNTEyMjcxOVowIDEeMBwGA1UEAwwVdGxzd3JhcHBlcl90ZXN0
+X290aGVyMIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEA0V05f4mWoxcW
+WriaRN8spGxRP6Nk2Jz0F9y3JX06F7+kUCAD4xr8K7wYB+JtSPT2mGKZx6JZ8RGA
+zbq9FSbrX5MqXVyivTjXFSMe+jffM9oQQN7uly0pqK17vMLB84EgYqBAdW4xGJL6
+N7WaKErQ4L0f4IQrjlySOCvKo3VHfk+FYVlHj+qktypQDsVasLa9lOWjd4KSRpWU
+LpCcpFI0JEKdY3KFtdxJmBZxNqUY9zXt/kyw01XjNrJkH+XcC/3mnrFOmHPjIvEP
+BrTMR+Yc09eG/8RtxXOZxj1CuCrfCJf/ja8vdFBshVYZ2y3HpeItUofEFy2WT3gh
+jGiUgpwfuwIDAQABo1MwUTAdBgNVHQ4EFgQU947UQygfsOw+udnjR3rywSlqnZIw
+HwYDVR0jBBgwFoAU947UQygfsOw+udnjR3rywSlqnZIwDwYDVR0TAQH/BAUwAwEB
+/zANBgkqhkiG9w0BAQsFAAOCAQEAZUrhuONUlLalvmtjaVEsNgoPErF8jPeLHHGA
+9dVAvkaumbwacO5AYWbODRhC0NCdYTNAqLaQVA4utxZGI3Z2pQKxI+OnlCM+ws3z
+Nrfh+E/6EJZblfqK7SBduPYlFxhkalQ19mi/R5E6p9EmBdV1sScR6Hsg3qSCpFbT
+Z49kghXJ5KAS4jOB6SxClxbR5Tpc1E3khduX6aGau1VOkgPJxfdqHHqsyUc1RH/Z
+3W6n4SmhCZxEpQzSQUw4YPIIpWuSUWUr7MS7TzGDdT4AgD0m5VLFbF2ca/Fsz2WW
+v2R63aFo/UfQSQ4dhC0o2Vy74DyLwnO3pH8wudfBJ8/LX/Uz/A==
+-----END CERTIFICATE-----)";
+
+constexpr auto kShortTimeout = std::chrono::milliseconds{10};
+
+inline constexpr std::chrono::seconds kMaxWaitTime(60);
+}  // namespace
+
+// Macros for generating an initializer lists for tests WriteAll
+// - begin
+#define MSG "msg"
+#define IO_Data \
+  { MSG, 3 }
+#define IO_Data_5 IO_Data, IO_Data, IO_Data, IO_Data, IO_Data
+#define IO_Data_25 IO_Data_5, IO_Data_5, IO_Data_5, IO_Data_5, IO_Data_5
+#define IO_Data_125 IO_Data_25, IO_Data_25, IO_Data_25, IO_Data_25, IO_Data_25
+#define IO_Data_625 \
+  IO_Data_125, IO_Data_125, IO_Data_125, IO_Data_125, IO_Data_125
+#define IO_Data_3125 \
+  IO_Data_625, IO_Data_625, IO_Data_625, IO_Data_625, IO_Data_625
+#define IO_Data_15625 \
+  IO_Data_3125, IO_Data_3125, IO_Data_3125, IO_Data_3125, IO_Data_3125
+// - end
+
+[[maybe_unused]] void tls_write_all_buffered(benchmark::State& state) {
+  for ([[maybe_unused]] auto _ : state) {
+    engine::RunStandalone(2, [&]() {
+      const auto deadline = Deadline::FromDuration(kMaxWaitTime);
+      internal::net::TcpListener listener;
+      auto [server, client] = listener.MakeSocketPair(deadline);
```

Comment:
All the code starting form `engine::RunStandalone(2, [&]() ` and ending with this line  should not be part of the `for ([[maybe_unused]] auto _ : state) { `